### PR TITLE
UI: Created Home Page | Issue - #4

### LINKED ISSUE #4: UI: Create Home Page
Add options for:

- [x] Selling a book (Floating Action Button)

- [x] Searching for books in store

### PR DESCRIPTION
**Pull request by unnikrishnan2002**
- Floating Action button added with book selling option
- Search bar added at the top of the page

### CODE REVIEW

File: .idea/runConfigurations.xml [hunk 1]
```diff
@@ -1,10 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Mustafa2856) [inline .idea/runConfigurations.xml:1]: restore this file

File: app/src/main/java/com/aosp/bookstore/MainActivity.java [hunk 2]
```diff
@@ -2,12 +2,61 @@
...
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
 
 public class MainActivity extends AppCompatActivity {
 
+    FloatingActionButton expand;
+    FloatingActionButton sellBook;
+
```
Comment (Mustafa2856) [inline app/src/main/java/com/aosp/bookstore/MainActivity.java:15]: Don't add unnecessary newlines

File: app/src/main/java/com/aosp/bookstore/MainActivity.java [hunk 3]
```diff
@@ -2,12 +2,61 @@
...
+    TextView text;
+
```
Comment (Mustafa2856) [inline app/src/main/java/com/aosp/bookstore/MainActivity.java]: Don't add unnecessary empty lines between variable declarations.

File: .idea/runConfigurations.xml (hunk 1 above)
Comment (Mustafa2856) [inline .idea/runConfigurations.xml:1]: Restore this file

File: app/src/main/java/com/aosp/bookstore/MainActivity.java [hunk 4]
```diff
@@ -2,12 +2,61 @@
...
+        });
+
+        sellBook.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Toast.makeText(MainActivity.this, "Clicked!!", Toast.LENGTH_SHORT).show();
+            }
+        });
     }
 }
```
Comment (Mustafa2856) [inline app/src/main/java/com/aosp/bookstore/MainActivity.java]: add newlines after each file

File: app/src/main/java/com/aosp/bookstore/MainActivity.java [hunk 5]
```diff
@@ -2,12 +2,61 @@
...
+
+                    sellBook.hide();
+                    text.setVisibility(View.GONE);
+
+                    isAllFabVisible = false;
+                }
+            }
+        });
+
+        sellBook.setOnClickListener(new View.OnClickListener() {
```
Comment (Mustafa2856) [inline app/src/main/java/com/aosp/bookstore/MainActivity.java]: use lambda functions for anonymous calls

File: app/src/main/java/com/aosp/bookstore/MainActivity.java [hunk 6]
```diff
@@ -2,12 +2,61 @@
...
+        sellBook = findViewById(R.id.sellBook);
+
+        text = findViewById(R.id.text);
+
+        sellBook.setVisibility(View.GONE);
+        text.setVisibility(View.GONE);
+
+        isAllFabVisible = false;
+
+        expand.setOnClickListener(new View.OnClickListener() {
```
Comment (Mustafa2856) [inline app/src/main/java/com/aosp/bookstore/MainActivity.java]: use lambda functions for anonymous calls

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,24 +1,85 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout
-        xmlns:android="http://schemas.android.com/apk/res/android"
-        xmlns:tools="http://schemas.android.com/tools"
-        xmlns:app="http://schemas.android.com/apk/res-auto"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        tools:context=".MainActivity">
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (Mustafa2856) [inline app/src/main/res/layout/activity_main.xml]: unnecessary change in syntax for first 10 lines

File: app/src/main/res/layout/activity_main.xml [hunk 8]
```diff
@@ -1,24 +1,85 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
 
     <TextView
```
Comment (Mustafa2856) [inline app/src/main/res/layout/activity_main.xml]: the welcome text was place holder, no longer needed, add a scroll view here to show search results

File: app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -1,24 +1,85 @@
...
+        android:layout_marginLeft="25dp"
+        android:layout_marginTop="40dp"
+        android:layout_marginRight="25dp"
+        android:background="@drawable/search_bg"
+        android:drawableStart="@drawable/ic_baseline_search_24"
+        android:ems="10"
+        android:inputType="textPersonName"
+        android:padding="16dp"
+        android:paddingLeft="16dp"
+        android:hint="  Search..."
```
Comment (Mustafa2856) [inline app/src/main/res/layout/activity_main.xml]: make hint to be "search books available"

**Review by Mustafa2856 — CHANGES_REQUESTED** [review]

**Comment by unnikrishnan2002** [comment]
@Mustafa2856 All the requested changes have been made, added the runConfigurations.xml file, and **I had to change the versions of some dependencies because it was throwing errors otherwise**

**Review by Mustafa2856 — APPROVED** [review]

**Pull request merged**